File: cpfed/506992/c,d.py
# https://cpfedcoach24.contest.codeforces.com/group/Y0bboME4W4/contest/506992/problem/C
# https://cpfedcoach24.contest.codeforces.com/group/Y0bboME4W4/contest/506992/problem/D

import sys

# Read n from input
n = int(input())

l = 1
r = n
while r - l > 1:
    x = int((r + l) / 2)
    print(x, flush = True)
    if input() == "<":
        r = x - 1
    else:
        l = x

# Write answer
print(r, flush = True)
if input() == ">=":
    print("! " + str(r), flush = True)
else:
    print("! " + str(l), flush = True)

File: cpfed/506992/c,d.cpp
/**
 * https://cpfedcoach24.contest.codeforces.com/group/Y0bboME4W4/contest/506992/problem/C
 * https://cpfedcoach24.contest.codeforces.com/group/Y0bboME4W4/contest/506992/problem/D
 * @author Arman Sydikov
 */
#include <bits/stdc++.h>

using namespace std;

int main() {
    // fast input
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    // input
    int n;
    cin >> n;

    int left = 1;
    int right = n;
    while (left < right) {
        int middle = (left + right) / 2 + 1;
        cout << middle << endl;
        fflush(stdout);

        string s;
        cin >> s;
        if (s == "<") {
            right = middle-1;
        } else {
            left = middle;
        }
    }

    cout << "! " << left;
}